### PR TITLE
add origin openapi aggregation

### PR DESCRIPTION
**Pull request by juanvallejo**
OpenAPI aggregation wiring discussed during kubecon.
Demonstrates a simpler approach to https://github.com/openshift/origin/pull/21448
~~Currently fails e2e tests due to breaking some existing endpoints.~~

cc @sttts @soltysh

### CODE REVIEW

File: pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go [hunk 1]
```diff
@@ -509,6 +511,28 @@ func (c *completedConfig) withUserAPIServer(delegateAPIServer genericapiserver.D
 	return server.GenericAPIServer, &legacyStorageVersionMutator{version: userapiv1.SchemeGroupVersion, storage: storage}, nil
 }
 
+func (c *completedConfig) withOpenAPIAggregator(delegatedAPIServer *genericapiserver.GenericAPIServer) (*genericapiserver.GenericAPIServer, error) {
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go]: withOpenAPIAggregationController. Also remove the first return value. This is not functional, but mutates the input. Let's make that clear.

File: pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go [hunk 2]
```diff
@@ -509,6 +511,28 @@ func (c *completedConfig) withUserAPIServer(delegateAPIServer genericapiserver.D
...
+	delegatedAPIServer.RemoveOpenAPIData()
```
Comment (sttts) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go]: add a comment why this is needed and why this is sane. Why don't we lose any important openapi specs this way from the passed API server?

**Comment by sttts** [comment]
With the cosmetical issues fixed, this looks good!

File: pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go]: > Why don't we lose any important openapi specs this way from the passed API server?

As far as I've been able to tell, we instantiate a "default" openapi config here: https://github.com/openshift/origin/blob/master/pkg/cmd/openshift-apiserver/openshiftapiserver/config.go#L107

**Comment by juanvallejo** [comment]
@sttts thanks for the review, comments addressed

**Comment by soltysh** [comment]
/hold
until after the rebase

**Pull request merged**